### PR TITLE
Revise CertificateSigningRequest documentation

### PR DESCRIPTION
**Pull request by sftim**
~WIP~ Tidying following PR #19290

Motivation: https://github.com/kubernetes/website/pull/19290#issuecomment-600078064

### CODE REVIEW

**Comment by zacharysarah** [comment]
@sftim This looks like it's headed in the right direction. Thanks for taking on the follow-up! 🙇 

/approve

Did you still want to address whether this page belongs in [concepts vs. references](https://github.com/kubernetes/website/pull/19290#discussion_r391906214)? 

For purposes of discussing page location (feel free to remove once resolved):
/hold

Leaving LGTM for @liggitt @munnerz @deads2k

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 1]
```diff
@@ -25,38 +26,39 @@ finally being signed.
...
 Once created, a CertificateSigningRequest must be approved before it can be signed.
-Depending on the signer requested, a CertificateSigningRequest may be auto-approved by a
-{{< glossary_tooltip text="controller" term_id="controller" >}}.
+Depending on the signer selected, a CertificateSigningRequest may be automatically approved
+by a {{< glossary_tooltip text="controller" term_id="controller" >}}.
 Otherwise, a CertificateSigningRequest must be manually approved either via the REST API (or client-go)
-or via `kubectl certificate approve`. Likewise, a CertificateSigningRequest may also be denied, in which
-case the configured signer must not sign the request.
+or by running `kubectl certificate approve`. Likewise, a CertificateSigningRequest may also be denied,
+which tells the configured signed that it must not sign the request.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
which tells the configured signer that it must not sign the request.
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 2]
```diff
@@ -25,38 +26,39 @@ finally being signed.
...
-Once approved, a signing controller must attempt to sign the request. A signer should apply its own policy
-restrictions, similar to the approval process, on the request to ensure it never signs an invalid request.
-There is currently not a mechanism for a signer implementation to report its inability to sign a request.
-Once the signer has determined that a request should be signed, it must sign the request and store the
-resulting certificate (and if appropriate, certificate chain) in the `status.certificate` field as PEM-encoded
-certificates.
+For certificates that have been approved, the next step is signing. The relevant signing controller
+first validates that the signing conditions are met and then creates a certificate.
+The signing controller then updates the CertificateSigningRequest, storing the new certificate into
+the the `status.certificate` field of existing CertificateSigningReques object. The
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
the the `status.certificate` field of the existing CertificateSigningReques object. The
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 3]
```diff
@@ -25,38 +26,39 @@ finally being signed.
...
+`status.certificate` field is either empty or contains a X.509 certificatem, encoded in PEM format.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: I don't understand the statement about it being empty in the section about the signer populating it.

Comment (sftim) [reply content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: I'm trying to convey that you can have a Pending / Denied CSR and if you do then there won't be a certificate.

Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: Ah, maybe state that separately then?

```
The signing controller then updates the CertificateSigningRequest, storing the new certificate into
the `status.certificate` field of the existing CertificateSigningRequest object, encoded in PEM format.
The CertificateSigningRequest `status.certificate` field is empty until the signer does this.
```

Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
`status.certificate` field is either empty or contains a X.509 certificate, encoded in PEM format.
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 4]
```diff
@@ -65,8 +67,9 @@ This includes:
...
 1. **Expiration/certificate lifetime**: whether it is fixed by the signer, configurable by the admin, determined by the CSR object etc and behavior if an expiration different than the signer-determined expiration is specified in the CSR.
-1. **CA bit allowed/disallowed**: and behavior if a CSR contains a request a for a CA cert when the signer does not permit it.
-1. **optional**: Information about the meaning of additional `CERTIFICATE` PEM blocks in `status.certificate`, if different from the standard behavior of treating the additional certificates as intermediates, and presenting them in TLS handshakes.
+1. **CA bit allowed/disallowed**: and behavior if a CSR contains a request a for a CA certificate when the signer does not permit it.
+
+Commonly, the `status.certificate` field contains a single PEM-encoded X.509 certificate once the CSR is approved and the certificate is issued. Some signers store multiple certificates into the `status.certificate` field. In that case, the documentation for the signer should specify the meaning of additional certificates; for example, this might be certificate plus intermediates to be presented during TLS handshakes.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: this should convey that the default meaning of additional certificates is that they are an intermediate chain

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 5]
```diff
@@ -78,7 +81,7 @@ Kubernetes provides built-in signers that each have a well-known `signerName`:
     1. Permitted subjects - no subject restrictions, but approvers and signers may choose not to approve or sign. Certain subjects like cluster-admin level users or groups vary between distributions and installations, but deserve additional scrutiny before approval and signing. The `CertificateSubjectRestriction` admission plugin is available and enabled by default to restrict `system:masters`, but it is often not the only cluster-admin subject in a cluster.
     1. Permitted x509 extensions - honors subjectAltName and key usage extensions and discards other extensions.
     1. Permitted key usages - must include []string{"client auth"}. Must not include key usages beyond []string{"digital signature", "key encipherment", "client auth"}
-    1. Expiration/cert lifetime - minimum of CSR signer or request. Sanity of the time is the concern of the signer.
+    1. Expiration/certificate lifetimeime - minimum of CSR signer or request. The signer is responsible for checking that the certificate lifetime is valid and permissible.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
    1. Expiration/certificate lifetime - minimum of CSR signer or request. The signer is responsible for checking that the certificate lifetime is valid and permissible.
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 6]
```diff
@@ -89,7 +92,7 @@ Kubernetes provides built-in signers that each have a well-known `signerName`:
     1. Permitted subjects - organizations are exactly `[]string{"system:nodes"}`, common name starts with `"system:node:"`
     1. Permitted x509 extensions - honors key usage extensions, forbids subjectAltName extensions, drops other extensions.
     1. Permitted key usages - exactly `[]string{"key encipherment", "digital signature", "client auth"}`
-    1. Expiration/cert lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
+    1. Expiration/certificate lifetimeime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
    1. Expiration/certificate lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 7]
```diff
@@ -100,17 +103,17 @@ Kubernetes provides built-in signers that each have a well-known `signerName`:
     1. Permitted subjects - organizations are exactly `[]string{"system:nodes"}`, common name starts with `"system:node:"`
     1. Permitted x509 extensions - honors key usage and DNSName/IPAddress subjectAltName extensions, forbids EmailAddress and URI subjectAltName extensions, drops other extensions. At least one DNS or IP subjectAltName must be present.
     1. Permitted key usages - exactly `[]string{"key encipherment", "digital signature", "server auth"}`
-    1. Expiration/cert lifetime - minimum of CSR signer or request.
+    1. Expiration/certificate lifetimeime - minimum of CSR signer or request.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
    1. Expiration/certificate lifetime - minimum of CSR signer or request.
```

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md [hunk 8]
```diff
@@ -100,17 +103,17 @@ Kubernetes provides built-in signers that each have a well-known `signerName`:
...
-  certs, but that behavior is not standard kubernetes behavior.
+  certs, but that behavior is not standard Kubernetes behavior.
   Never auto-approved by {{< glossary_tooltip term_id="kube-controller-manager" >}}.
-    1. Trust distribution: None.  There is no standard trust or distribution for this signer in a kubernetes cluster.
+    1. Trust distribution: None.  There is no standard trust or distribution for this signer in a Kubernetes cluster.
     1. Permitted subjects - any
     1. Permitted x509 extensions - honors subjectAltName and key usage extensions and discards other extensions.
     1. Permitted key usages - any
-    1. Expiration/cert lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
+    1. Expiration/certificate lifetimeime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]:
```suggestion
    1. Expiration/certificate lifetime - minimum of CSR signer or request.  Sanity of the time is the concern of the signer.
```

**Comment by liggitt** [comment]
a few nits, lgtm otherwise

**Comment by VineethReddy02** [comment]
Hi @sftim @liggitt 
As we are very close to 1.18 release. I would appreciate it if we can get this done the earliest.

Thanks!

**Comment by sftim** [comment]
@VineethReddy02 noted. I should be able to put time into this in the next 12 hours.

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: To document this, I need a question answered: If there is more than one certificate, what's the convention for ordering?

For example:
- The first is the server's / clients certificate and the others chain onwards towards the root?
- The first is the intermediate certificate signed by the root,  then the next intermediate CA, and so on. The last in the list is the server's / client's certificate.
- The certificates are in any order; the TLS implementation should solve and find a valid chain then present that to a peer?
- something else?

**Comment by sftim** [comment]
If v1.18 comes out, I'll cherry-pick the same commits onto _master_, force-push, and change the base branch. Not a big problem in my view.

(if you're reading this and you think it's important to get these changes into _dev-1.18_ before release, please reply and I'll bear that in mind).

**Comment by sftim** [comment]
I also plan to add a new page something like https://kubernetes.io/docs/concepts/security/certificate-signing/
I think that's also something that can land on _master_ after the v1.18 release comes out.

File: content/en/docs/reference/access-authn-authz/certificate-signing-requests.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/certificate-signing-requests.md]: Think this is sorted.

**Pull request merged**